hpedockerplugin/volume_manager.py

```diff
@@ -1101,6 +1101,8 @@ def get_volume_snap_details(self, volname, snapname, qualified_name):
             vol_detail['cpg'] = volinfo.get('cpg')
             vol_detail['snap_cpg'] = volinfo.get('snap_cpg')
             vol_detail['backend'] = volinfo.get('backend')
+            vol_detail['domain'] = self._hpeplugin_driver.get_domain(
```

Comment:
@imran-ansari can we introduce domain field in volume.py? This will reduce the number of 3PAR calls.
The second opinion is for the older volumes which are created via 2.1, after an upgrade to 3.0 can we use existing get_cpg and get_snapcpg calls to get domain information as well?
This will ensure that there are minimum 3PAR calls while inspecting.

Reply:
The reason not to include domain as part of volume metadata was to avoid changes in multiple places.
In my opinion, if we really see a slowdown in inspect operation due to calls to array then only we should go for optimization. I'm okay to make the changes though.

Reply:
@imran-ansari As long as there are no issues reported with the performance from an array while querying the domain, this change is okay for me to go in.